web-ui/src/main/java/org/archcnl/ui/input/ConceptAndRelationView.java

```diff
@@ -13,22 +19,68 @@
     CreateNewLayout createNewRelationLayout =
             new CreateNewLayout("Relations", "Create new relation");
     HorizontalLayout bottomBarLayout = new HorizontalLayout();
-
-    public void setUpBottomBar() {
-        final Button saveButton = new Button("Save");
-        final Button checkButton = new Button("Check");
-        bottomBarLayout.add(saveButton, checkButton);
-    }
+    TreeGrid<ListEntry<Concept>> conceptTreeGrid = new TreeGrid<>();
+    TreeGrid<ListEntry<Relation>> relationTreeGrid = new TreeGrid<>();
 
     public ConceptAndRelationView() {
         createNewConceptLayout.setHeight(50, Unit.PERCENTAGE);
         createNewRelationLayout.setHeight(50, Unit.PERCENTAGE);
 
         setUpBottomBar();
+        setUpConceptView();
+        setUpRelationView();
 
         add(createNewConceptLayout);
         add(createNewRelationLayout);
         add(bottomBarLayout);
         getStyle().set("border", "1px solid black");
+
+        createNewConceptLayout.add(conceptTreeGrid);
+        createNewRelationLayout.add(relationTreeGrid);
+    }
+
+    public void setUpBottomBar() {
+        final Button saveButton = new Button("Save");
+        final Button checkButton = new Button("Check");
+        bottomBarLayout.add(saveButton, checkButton);
+    }
+
+    public void setUpConceptView() {
+        conceptTreeGrid = new TreeGrid<>();
+
+        RulesConceptsAndRelations car = RulesConceptsAndRelations.getInstance();
+        List<Concept> manager = car.getConceptManager().getConcepts();
+
+        List<ListEntry<Concept>> data = new LinkedList<>();
+        ListEntry<Concept> defaultConceptsStub = new ListEntry<>("Default Concepts", manager);
+        data.add(defaultConceptsStub);
+
+        conceptTreeGrid.setItems(data, ListEntry::getHierarchicalChildren);
+        conceptTreeGrid.addComponentHierarchyColumn(
+                entry -> {
+                    ListEntryLayout<Concept> entryLayout = new ListEntryLayout<>(entry);
+                    entryLayout.setSizeFull();
+                    return entryLayout;
+                });
+    }
+
+    private void setUpRelationView() {
+        relationTreeGrid = new TreeGrid<>();
+
+        RulesConceptsAndRelations car = RulesConceptsAndRelations.getInstance();
```

Comment:
The variable name `car` is misleading. Arguably the variable is not necessary, to begin with as it is only used once.

Reply:
Variable has been removed.